examples/localstack/localstack.go

```diff
@@ -0,0 +1,150 @@
+package localstack
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/internal/testcontainersdocker"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"golang.org/x/mod/semver"
+)
+
+const defaultPort = 4566
+const defaultRegion = "us-east-1"
+const defaultVersion = "0.11.2"
+const hostnameExternalEnvVar = "HOSTNAME_EXTERNAL"
+
+const defaultAccessKeyID = "accesskey"
+const defaultSecretAccessKey = "secretkey"
+const defaultToken = "token"
+
+func runInLegacyMode(image string) bool {
+	parts := strings.Split(image, ":")
+	version := parts[len(parts)-1]
+
+	if version == "latest" {
+		return false
+	}
+
+	if !strings.HasPrefix(version, "v") {
+		version = fmt.Sprintf("v%s", version)
+	}
+
+	if semver.IsValid(version) {
+		return semver.Compare(version, "v0.11") < 0 // version < v0.11
+	}
+
+	fmt.Printf("Version %s is not a semantic version, LocalStack will run in legacy mode.\n", version)
+	fmt.Printf("Consider using \"StartContainer(ctx, NoopOverrideContainerRequest, WithLegacyMode)\" constructor if you want to enable legacy mode.")
```

Comment:
I think those message are confusing a little bit. In line 37, my understanding is if version is a valid version and less than v0.11 then it will run in legacy mode. So, the print messages should be 

```suggestion
	fmt.Printf("Version %s is not a semantic version, LocalStack will not run in legacy mode.\n", version)
	fmt.Printf("Consider using \"StartContainer(ctx, NoopOverrideContainerRequest, WithLegacyMode)\" constructor if you want to enable legacy mode.")
```

Please, correct me if I am wrong

Reply:
I took them from Java: https://github.com/testcontainers/testcontainers-java/blob/main/modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java#L112